sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/ServiceBusHostBuilderExtensions.cs

```diff
@@ -110,6 +110,7 @@ public static IWebJobsBuilder AddServiceBus(this IWebJobsBuilder builder, Action
 
             builder.Services.AddAzureClientsCore();
             builder.Services.TryAddSingleton<MessagingProvider>();
+            builder.Services.AddHostedService<CachedClientCleanupService>();
```

Comment:
Do we need another service to do this cleanup?  Can we not update MesasgingProvider to implement IDisposable? And perform this cleanup in the dispose implementation?

Reply:
We don't want to make MessagingProvider disposable because it is public and it could cause compiler warnings for customers that could end up being build errors depending on their configuration. We can't make it be a HostedService because that changes the pattern for how the service needs to be added which could also break customers using a custom MessagingProvider.